### ISSUE TITLE
[feature request] Integrate SuperGrubDisk2 functionnality as a tool in ventoy

### ISSUE THREAD

**Issue by mhermier** [ventoy/Ventoy#693]
Hi,
I think it would be a nice feature to augment ventoy so that it is also able to boot system directly. Since ventoy is grub based as SuperGrubDisk2 (and looking at its code which is mostly(only?) grub scripts), I think it can be doable for "free". Integrating it, could also help to be able to remove the noisy entries exposed by ventoy internals.

**Comment by ventoy** [comment]
I think it's more reasonable to boot SuperGrubDisk2 iso file from Ventoy not integrate it.
After all, they are two different sulotions.

**Comment by mhermier** [comment]
I do not really agree, they are both ment to boot a system on a medium. But
even without that, maybe viability can be tested as a plug-in first? I
don't know the plug-in infrastructure to know if a grub script can be ran
from ventoy.

**Comment by ventoy** [comment]
Not exactly.
Ventoy just create a virtual cdrom device and chainload to it. 
So the system is loaded by the bootloader inside the ISO file(e.g. isolinux/grub/lilo...). Ventoy doesn't load the system directly with grub2's  linux/initrd cmd as SuperGrubDisk2 does.

Grub2 is just the frontend of Ventoy In a sense.

**Comment by mhermier** [comment]
The virtualization is an implementation detail (that is the specificity of
ventoy), but I think grub does the work of chain load I guess. It should
mean it was modified just enough to administer the virtualized device. So
it should be able to run grub regular scripts as expected (minor the
removed support that was not expected/required). Or am I mistaken somewhere?

**Comment by mhermier** [comment]
I did a preliminary test in VM using "Menu Extension Plugin" and it trivially worked with some expected integration/user experience issue.
I have to make some tests to see if its viable for my odd configuration (requires raid modules) but it seems really promising.

**Comment by ventoy** [comment]
Yes. Ventoy just add some extension for Grub2. All grub2's original commands are avaliable.
You can add your custom grub2 script as you want with F6 "Menu Extension Plugin". 
But that's not Ventoy's way and will not included in the official ventoy release.

**Comment by mhermier** [comment]
Even if if it not ventoy way, I would like to ask about the ease of
deployment of such tools/script packs. It would be nice to be able to put a
directory somewhere in /ventoy and that a menu was somehow automagically
filled with an entry point without human interaction. But maybe I'm asking
for too much... I'll try to make some fork of super grub so it can generate
a module for ventoy soon.

**Comment by mhermier** [comment]
I looked at the code and unless I'm miss-taken there is no command to query a json, could it be added ? Something like "vt_json_get $file $node_name" it would help to create a meta script, I can use other tricks, but it would feel less integrated.

**Comment by ventoy** [comment]
If you make a fork super grub2 disk. I think it's easier to make it work better with Ventoy than integrate it into Ventoy.

SuperGrub2Disk/agFM ... and other similar software offer bootable ISO files. It's more reasonable to just put their ISO files to Venoty USB stick and boot them from Ventoy when needed.

**Comment by mhermier** [comment]
I have an *odd* usage of those tools (I use them as my primary boot loader
to raid whole disc), and it is bothering me basically to waste memory to
boot 2 systems basically to have 2 grub environment running, while one is
enough.

So far the plan changed to integrating, to make a 2 plug-in system. One
plug-in to load more of them, because only one menu module can exist. And
the real supergrub2 one. This is already working, I struggle a little
understand exiting modules, to makes things a little bit more user
friendly. It is rough around the edges. But should append sooner or later.
